File: core/internal/server/organizations.go
package server

type OrganizationStartMode string

const (
	OrganizationStartModeTemplate OrganizationStartMode = "template"
	OrganizationStartModeEmpty    OrganizationStartMode = "empty"
)

type OrganizationAIEngineProfileID string
type ResponseContractProfileID string
type OrganizationOutputModelRoutingMode string
type OrganizationOutputTypeID string

const (
	OrganizationAIEngineProfileStarterDefaults OrganizationAIEngineProfileID = "starter_defaults"
	OrganizationAIEngineProfileBalanced        OrganizationAIEngineProfileID = "balanced"
	OrganizationAIEngineProfileHighReasoning   OrganizationAIEngineProfileID = "high_reasoning"
	OrganizationAIEngineProfileFastLightweight OrganizationAIEngineProfileID = "fast_lightweight"
	OrganizationAIEngineProfileDeepPlanning    OrganizationAIEngineProfileID = "deep_planning"
)

const (
	OrganizationOutputModelRoutingModeSingleModel         OrganizationOutputModelRoutingMode = "single_model"
	OrganizationOutputModelRoutingModeDetectedOutputTypes OrganizationOutputModelRoutingMode = "detected_output_types"
)

const (
	OrganizationOutputTypeGeneralText       OrganizationOutputTypeID = "general_text"
	OrganizationOutputTypeResearchReasoning OrganizationOutputTypeID = "research_reasoning"
	OrganizationOutputTypeCodeGeneration    OrganizationOutputTypeID = "code_generation"
	OrganizationOutputTypeVisionAnalysis    OrganizationOutputTypeID = "vision_analysis"
)

const (
	ResponseContractProfileClearBalanced        ResponseContractProfileID = "clear_balanced"
	ResponseContractProfileStructuredAnalytical ResponseContractProfileID = "structured_analytical"
	ResponseContractProfileConciseDirect        ResponseContractProfileID = "concise_direct"
	ResponseContractProfileWarmSupportive       ResponseContractProfileID = "warm_supportive"
)

type organizationAIEngineProfile struct {
	ID          OrganizationAIEngineProfileID
	Summary     string
	Description string
	BestFor     string
}

type responseContractProfile struct {
	ID        ResponseContractProfileID
	Summary   string
	ToneStyle string
	Structure string
	Verbosity string
	BestFor   string
}

type OrganizationAgentTypeProfileSummary struct {
	ID                                 string `json:"id"`
	Name                               string `json:"name"`
	HelpsWith                          string `json:"helps_with"`
	AIEngineBindingProfileID           string `json:"ai_engine_binding_profile_id,omitempty"`
	AIEngineEffectiveProfileID         string `json:"ai_engine_effective_profile_id,omitempty"`
	AIEngineEffectiveSummary           string `json:"ai_engine_effective_summary"`
	InheritsDepartmentAIEngine         bool   `json:"inherits_department_ai_engine"`
	ResponseContractBindingProfileID   string `json:"response_contract_binding_profile_id,omitempty"`
	ResponseContractEffectiveProfileID string `json:"response_contract_effective_profile_id,omitempty"`
	ResponseContractEffectiveSummary   string `json:"response_contract_effective_summary"`
	InheritsDefaultResponseContract    bool   `json:"inherits_default_response_contract"`
	OutputTypeID                       string `json:"output_type_id,omitempty"`
	OutputTypeLabel                    string `json:"output_type_label,omitempty"`
	OutputModelEffectiveID             string `json:"output_model_effective_id,omitempty"`
	OutputModelEffectiveSummary        string `json:"output_model_effective_summary,omitempty"`
	InheritsDefaultOutputModel         bool   `json:"inherits_default_output_model"`
}

type OrganizationOutputModelBinding struct {
	OutputTypeID    string `json:"output_type_id"`
	OutputTypeLabel string `json:"output_type_label"`
	ModelID         string `json:"model_id,omitempty"`
	ModelSummary    string `json:"model_summary"`
}

type OrganizationOutputModelCatalogEntry struct {
	ModelID        string   `json:"model_id"`
	Label          string   `json:"label"`
	Summary        string   `json:"summary"`
	OutputTypeIDs  []string `json:"output_type_ids,omitempty"`
	ProviderID     string   `json:"provider_id,omitempty"`
	Installed      bool     `json:"installed"`
	Popular        bool     `json:"popular"`
	SelfHostable   bool     `json:"self_hostable"`
	HostingFit     string   `json:"hosting_fit,omitempty"`
	PopularityNote string   `json:"popularity_note,omitempty"`
	Source         string   `json:"source,omitempty"`
}

type OrganizationOutputModelReviewCandidate struct {
	OutputTypeID    string   `json:"output_type_id"`
	OutputTypeLabel string   `json:"output_type_label"`
	ModelID         string   `json:"model_id"`
	ModelSummary    string   `json:"model_summary"`
	Installed       bool     `json:"installed"`
	ReviewCriteria  []string `json:"review_criteria"`
}

type OrganizationOutputModelRoutingPayload struct {
	RoutingMode                string                                   `json:"routing_mode"`
	DefaultModelID             string                                   `json:"default_model_id,omitempty"`
	DefaultModelSummary        string                                   `json:"default_model_summary"`
	Bindings                   []OrganizationOutputModelBinding         `json:"bindings,omitempty"`
	AvailableModels            []OrganizationOutputModelCatalogEntry    `json:"available_models,omitempty"`
	RecommendedModels          []OrganizationOutputModelCatalogEntry    `json:"recommended_models,omitempty"`
	ReviewCandidates           []OrganizationOutputModelReviewCandidate `json:"review_candidates,omitempty"`
	HardwareSummary            string                                   `json:"hardware_summary"`
	ReviewPermissionPrompt     string                                   `json:"review_permission_prompt"`
	AutomaticSelectionCriteria []string                                 `json:"automatic_selection_criteria,omitempty"`
}

type OrganizationDepartmentSummary struct {
	ID                           string                                `json:"id"`
	Name                         string                                `json:"name"`
	SpecialistCount              int                                   `json:"specialist_count"`
	AIEngineOverrideProfileID    string                                `json:"ai_engine_override_profile_id,omitempty"`
	AIEngineOverrideSummary      string                                `json:"ai_engine_override_summary,omitempty"`
	AIEngineEffectiveProfileID   string                                `json:"ai_engine_effective_profile_id,omitempty"`
	AIEngineEffectiveSummary     string                                `json:"ai_engine_effective_summary"`
	InheritsOrganizationAIEngine bool                                  `json:"inherits_organization_ai_engine"`
	AgentTypeProfiles            []OrganizationAgentTypeProfileSummary `json:"agent_type_profiles,omitempty"`
}

type OrganizationTemplateSummary struct {
	ID                       string                          `json:"id"`
	Name                     string                          `json:"name"`
	Description              string                          `json:"description"`
	OrganizationType         string                          `json:"organization_type"`
	TeamLeadLabel            string                          `json:"team_lead_label"`
	AdvisorCount             int                             `json:"advisor_count"`
	DepartmentCount          int                             `json:"department_count"`
	SpecialistCount          int                             `json:"specialist_count"`
	Departments              []OrganizationDepartmentSummary `json:"departments,omitempty"`
	AIEngineSettingsSummary  string                          `json:"ai_engine_settings_summary"`
	ResponseContractSummary  string                          `json:"response_contract_summary"`
	MemoryPersonalitySummary string                          `json:"memory_personality_summary"`
}

type OrganizationSummary struct {
	ID                        string                `json:"id"`
	Name                      string                `json:"name"`
	Purpose                   string                `json:"purpose"`
	StartMode                 OrganizationStartMode `json:"start_mode"`
	TemplateID                string                `json:"template_id,omitempty"`
	TemplateName              string                `json:"template_name,omitempty"`
	TeamLeadLabel             string                `json:"team_lead_label"`
	AdvisorCount              int                   `json:"advisor_count"`
	DepartmentCount           int                   `json:"department_count"`
	SpecialistCount           int                   `json:"specialist_count"`
	AIEngineProfileID         string                `json:"ai_engine_profile_id,omitempty"`
	AIEngineSettingsSummary   string                `json:"ai_engine_settings_summary"`
	ResponseContractProfileID string                `json:"response_contract_profile_id,omitempty"`
	ResponseContractSummary   string                `json:"response_contract_summary"`
	MemoryPersonalitySummary  string                `json:"memory_personality_summary"`
	OutputModelRoutingMode    string                `json:"output_model_routing_mode,omitempty"`
	DefaultOutputModelID      string                `json:"default_output_model_id,omitempty"`
	DefaultOutputModelSummary string                `json:"default_output_model_summary,omitempty"`
	Status                    string                `json:"status"`
}

type OrganizationHomePayload struct {
	OrganizationSummary
	Description         string                           `json:"description,omitempty"`
	Departments         []OrganizationDepartmentSummary  `json:"departments,omitempty"`
	OutputModelBindings []OrganizationOutputModelBinding `json:"output_model_bindings,omitempty"`
}

type TeamLeadGuidedAction string

const (
	TeamLeadGuidedActionPlanNextSteps         TeamLeadGuidedAction = "plan_next_steps"
	TeamLeadGuidedActionFocusFirst            TeamLeadGuidedAction = "focus_first"
	TeamLeadGuidedActionReviewSetup           TeamLeadGuidedAction = "review_setup"
	TeamLeadGuidedActionResumeRetainedPackage TeamLeadGuidedAction = "resume_retained_package"
)

type OrganizationCreateRequest struct {
	Name       string                `json:"name"`
	Purpose    string                `json:"purpose"`
	StartMode  OrganizationStartMode `json:"start_mode"`
	TemplateID string                `json:"template_id,omitempty"`
}

type TeamLeadGuidanceRequest struct {
	Action         TeamLeadGuidedAction `json:"action"`
	RequestContext string               `json:"request_context,omitempty"`
}

type OrganizationAIEngineUpdateRequest struct {
	ProfileID string `json:"profile_id"`
}

type DepartmentAIEngineUpdateRequest struct {
	ProfileID                   string `json:"profile_id,omitempty"`
	RevertToOrganizationDefault bool   `json:"revert_to_organization_default,omitempty"`
}

type AgentTypeAIEngineUpdateRequest struct {
	ProfileID      string `json:"profile_id,omitempty"`
	UseTeamDefault bool   `json:"use_team_default,omitempty"`
}

type AgentTypeResponseContractUpdateRequest struct {
	ProfileID                    string `json:"profile_id,omitempty"`
	UseOrganizationOrTeamDefault bool   `json:"use_organization_or_team_default,omitempty"`
}

type OrganizationOutputModelRoutingUpdateRequest struct {
	RoutingMode    string                                        `json:"routing_mode"`
	DefaultModelID string                                        `json:"default_model_id,omitempty"`
	Bindings       []OrganizationOutputModelBindingUpdateRequest `json:"bindings,omitempty"`
}

type OrganizationOutputModelBindingUpdateRequest struct {
	OutputTypeID           string `json:"output_type_id"`
	ModelID                string `json:"model_id,omitempty"`
	UseOrganizationDefault bool   `json:"use_organization_default,omitempty"`
}

type ResponseContractUpdateRequest struct {
	ProfileID string `json:"profile_id"`
}

type TeamLeadGuidanceResponse struct {
	Action             TeamLeadGuidedAction       `json:"action"`
	RequestLabel       string                     `json:"request_label"`
	Headline           string                     `json:"headline"`
	Summary            string                     `json:"summary"`
	PrioritySteps      []string                   `json:"priority_steps"`
	SuggestedFollowUps []string                   `json:"suggested_follow_ups"`
	ExecutionContract  *TeamLeadExecutionContract `json:"execution_contract,omitempty"`
}

File: interface/lib/organizations.ts
export type OrganizationStartMode = "template" | "empty";
export type OrganizationAIEngineProfileId = "starter_defaults" | "balanced" | "high_reasoning" | "fast_lightweight" | "deep_planning";
export type ResponseContractProfileId = "clear_balanced" | "structured_analytical" | "concise_direct" | "warm_supportive";
export type OrganizationOutputModelRoutingMode = "single_model" | "detected_output_types";
export type OrganizationOutputTypeId = "general_text" | "research_reasoning" | "code_generation" | "vision_analysis";
export type LoopActivityStatus = "success" | "warning" | "failed";

export interface OrganizationLoopActivityItem {
    id: string;
    name: string;
    last_run_at: string;
    status: LoopActivityStatus;
    summary: string;
}

export type OrganizationAutomationTriggerType = "scheduled" | "event_driven";
export type OrganizationLearningInsightStrength = "emerging" | "consistent" | "strong";

export interface OrganizationAutomationOutcomeItem {
    summary: string;
    occurred_at: string;
}

export interface OrganizationAutomationItem {
    id: string;
    name: string;
    purpose: string;
    trigger_type: OrganizationAutomationTriggerType;
    owner_label: string;
    status: LoopActivityStatus;
    watches: string;
    trigger_summary: string;
    recent_outcomes?: OrganizationAutomationOutcomeItem[];
}

export interface OrganizationLearningInsightItem {
    id: string;
    summary: string;
    source: string;
    observed_at: string;
    strength: OrganizationLearningInsightStrength;
}

export interface OrganizationAgentTypeProfileSummary {
    id: string;
    name: string;
    helps_with: string;
    ai_engine_binding_profile_id?: OrganizationAIEngineProfileId;
    ai_engine_effective_profile_id?: OrganizationAIEngineProfileId;
    ai_engine_effective_summary: string;
    inherits_department_ai_engine: boolean;
    response_contract_binding_profile_id?: ResponseContractProfileId;
    response_contract_effective_profile_id?: ResponseContractProfileId;
    response_contract_effective_summary: string;
    inherits_default_response_contract: boolean;
    output_type_id?: OrganizationOutputTypeId;
    output_type_label?: string;
    output_model_effective_id?: string;
    output_model_effective_summary?: string;
    inherits_default_output_model: boolean;
}

export interface OrganizationOutputModelBinding {
    output_type_id: OrganizationOutputTypeId;
    output_type_label: string;
    model_id?: string;
    model_summary: string;
}

export interface OrganizationOutputModelCatalogEntry {
    model_id: string;
    label: string;
    summary: string;
    output_type_ids?: OrganizationOutputTypeId[];
    provider_id?: string;
    installed: boolean;
    popular: boolean;
    self_hostable: boolean;
    hosting_fit?: string;
    popularity_note?: string;
    source?: string;
}

export interface OrganizationOutputModelReviewCandidate {
    output_type_id: OrganizationOutputTypeId;
    output_type_label: string;
    model_id: string;
    model_summary: string;
    installed: boolean;
    review_criteria: string[];
}

export interface OrganizationOutputModelRoutingPayload {
    routing_mode: OrganizationOutputModelRoutingMode;
    default_model_id?: string;
    default_model_summary: string;
    bindings?: OrganizationOutputModelBinding[];
    available_models?: OrganizationOutputModelCatalogEntry[];
    recommended_models?: OrganizationOutputModelCatalogEntry[];
    review_candidates?: OrganizationOutputModelReviewCandidate[];
    hardware_summary: string;
    review_permission_prompt: string;
    automatic_selection_criteria?: string[];
}

export interface OrganizationDepartmentSummary {
    id: string;
    name: string;
    specialist_count: number;
    ai_engine_override_profile_id?: OrganizationAIEngineProfileId;
    ai_engine_override_summary?: string;
    ai_engine_effective_profile_id?: OrganizationAIEngineProfileId;
    ai_engine_effective_summary: string;
    inherits_organization_ai_engine: boolean;
    agent_type_profiles?: OrganizationAgentTypeProfileSummary[];
}

export interface OrganizationTemplateSummary {
    id: string;
    name: string;
    description: string;
    organization_type: string;
    team_lead_label: string;
    advisor_count: number;
    department_count: number;
    specialist_count: number;
    ai_engine_profile_id?: OrganizationAIEngineProfileId;
    ai_engine_settings_summary: string;
    response_contract_profile_id?: ResponseContractProfileId;
    response_contract_summary: string;
    memory_personality_summary: string;
}

export interface OrganizationSummary {
    id: string;
    name: string;
    purpose: string;
    start_mode: OrganizationStartMode;
    template_id?: string;
    template_name?: string;
    team_lead_label: string;
    advisor_count: number;
    department_count: number;
    specialist_count: number;
    ai_engine_profile_id?: OrganizationAIEngineProfileId;
    ai_engine_settings_summary: string;
    response_contract_profile_id?: ResponseContractProfileId;
    response_contract_summary: string;
    memory_personality_summary: string;
    output_model_routing_mode?: OrganizationOutputModelRoutingMode;
    default_output_model_id?: string;
    default_output_model_summary?: string;
    status: string;
}

export interface OrganizationHomePayload extends OrganizationSummary {
    description?: string;
    departments?: OrganizationDepartmentSummary[];
    output_model_bindings?: OrganizationOutputModelBinding[];
}

export interface OrganizationCreateRequest {
    name: string;
    purpose: string;
    start_mode: OrganizationStartMode;
    template_id?: string;
}

export interface OrganizationAIEngineUpdateRequest {
    profile_id: OrganizationAIEngineProfileId;
}

export interface DepartmentAIEngineUpdateRequest {
    profile_id?: OrganizationAIEngineProfileId;
    revert_to_organization_default?: boolean;
}

export interface AgentTypeAIEngineUpdateRequest {
    profile_id?: OrganizationAIEngineProfileId;
    use_team_default?: boolean;
}

export interface AgentTypeResponseContractUpdateRequest {
    profile_id?: ResponseContractProfileId;
    use_organization_or_team_default?: boolean;
}

export interface ResponseContractUpdateRequest {
    profile_id: ResponseContractProfileId;
}

export interface OrganizationOutputModelBindingUpdateRequest {
    output_type_id: OrganizationOutputTypeId;
    model_id?: string;
    use_organization_default?: boolean;
}

export interface OrganizationOutputModelRoutingUpdateRequest {
    routing_mode: OrganizationOutputModelRoutingMode;
    default_model_id: string;
    bindings: OrganizationOutputModelBindingUpdateRequest[];
}

export type TeamLeadGuidedAction = "plan_next_steps" | "focus_first" | "review_setup";
export type TeamLeadExecutionMode = "guided_review" | "native_team" | "external_workflow_contract";

export interface TeamLeadGuidanceRequest {
    action: TeamLeadGuidedAction;
    request_context?: string;
}

export interface TeamLeadExecutionContract {
    execution_mode: TeamLeadExecutionMode;
    owner_label: string;
    summary: string;
    team_name?: string;
    external_target?: string;
    target_outputs: string[];
    workflow_group?: TeamLeadWorkflowGroupDraft;
}

export interface TeamLeadWorkflowGroupDraft {
    name: string;
    goal_statement: string;
    work_mode: "read_only" | "propose_only" | "execute_with_approval" | "execute_bounded";
    coordinator_profile: string;
    allowed_capabilities?: string[];
    expiry_hours?: number;
    summary: string;
}

export interface TeamLeadGuidanceResponse {
    action: TeamLeadGuidedAction;
    request_label: string;
    headline: string;
    summary: string;
    priority_steps: string[];
    suggested_follow_ups: string[];
    execution_contract?: TeamLeadExecutionContract;
}
